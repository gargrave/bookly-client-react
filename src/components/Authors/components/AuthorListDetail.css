div.author-list-detail {
  cursor: pointer;
  border: solid 1px #bbb;
  border-radius: 5px;
  box-shadow: 0 1px 3px rgba(0, 0, 0, .12), 0 1px 2px rgba(0, 0, 0, .24);
  margin: 10px auto;
  max-width: 600px;
  padding: 8px;
}

div.author-list-detail:hover {
  background: #ffe8e4;
}

div.author-list-detail p {
  margin: 0;
}
