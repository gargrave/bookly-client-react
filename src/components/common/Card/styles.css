.bookly-card {
  background-color: #f9f9f9;
  cursor: pointer;
  border: solid 1px #bbb;
  border-radius: 5px;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
  margin: 10px auto;
  width: 95%;
  max-width: 600px;
  padding: 8px;
}

.bookly-card:hover {
  background: #ffe8e4;
}

.bookly-card p {
  margin: 0;
}

.bookly-card .bookly-card-title {
  font-weight: bold;
}

.bookly-card .bookly-card-text {
  color: #6e6969;
}
