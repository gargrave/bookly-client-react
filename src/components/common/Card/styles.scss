.bookly-card {
  background-color: #f9f9f9;
  border: solid 1px #bbb;
  border-radius: 5px;
  box-shadow:
    0 1px 3px rgba(0, 0, 0, 0.07),
    0 1px 2px rgba(0, 0, 0, 0.12);
  margin: 8px 10px;
  padding: 10px 20px;
  width: 95%;

  hr {
    margin: 12px 0;
  }

  &.bookly-hoverable:hover {
    background: #ffe8e4;
    cursor: pointer;
  }

  p {
    margin: 0;
  }

  &-title {
    font-weight: bold;
  }

  &-text {
    color: #6e6969;
  }

  &-text + &-text {
    margin-top: 8px;
  }

  &-spacer {
    height: 20px;
    line-height: 20px;
  }

  &.bookly-detail-card {
    .bookly-card-title {
      font-size: 1.3rem;
      line-height: 2.5rem;
    }

    .bookly-card-title + .bookly-card-text {
      font-size: 1.1rem;
    }
  }
}
