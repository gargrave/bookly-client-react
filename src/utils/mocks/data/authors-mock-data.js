export const authorsMockData = [
  {
    id: 8,
    firstName: 'Aldous',
    lastName: 'Huxley',
    createdAt: '2017-07-01T01:16:43.881Z',
    updatedAt: '2017-07-01T01:16:43.881Z'
  },
  {
    id: 51,
    firstName: 'Kurt',
    lastName: 'Vonnegut',
    createdAt: '2017-07-06T22:02:11.504Z',
    updatedAt: '2017-07-06T22:02:11.504Z'
  },
  {
    id: 59,
    firstName: 'George R. R.',
    lastName: 'Martin',
    createdAt: '2017-08-17T20:47:35.232Z',
    updatedAt: '2017-08-17T20:47:35.232Z'
  },
  {
    id: 52,
    firstName: 'Tom',
    lastName: 'Robbins',
    createdAt: '2017-07-06T22:02:17.051Z',
    updatedAt: '2017-08-17T21:27:35.885Z'
  }
]
