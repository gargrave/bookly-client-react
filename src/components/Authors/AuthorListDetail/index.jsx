import React from 'react'
import { func, number, shape, string } from 'prop-types'

import './styles.css'

const AuthorListDetail = (props) => {
  return (
    <div className="author-list-detail" onClick={props.onClick}>
      <p className="author-name">
        {props.author.firstName} {props.author.lastName}
      </p>
    </div>
  )
}

AuthorListDetail.propTypes = {
  author: shape({
    id: number.isRequired,
    firstName: string.isRequired,
    lastName: string.isRequired,
    createdAt: string.isRequired,
    updatedAt: string.isRequired,
  }),
  onClick: func.isRequired,
}

export default AuthorListDetail
