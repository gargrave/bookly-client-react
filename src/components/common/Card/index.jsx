import React from 'react'
import { array, func, string } from 'prop-types'

import { buildClasses } from '@/utils/cssHelpers'

import './styles.css'

const renderText = (text, classname) => (
  text
    ? <p className={buildClasses(classname)}>{text}</p>
    : null
)

const Card = ({
  children,
  classes = [],
  onClick,
  text,
  title,
}) => {
  return (
    <div
      className={buildClasses(['card', ...classes])}
      onClick={onClick}>
      {renderText(title, 'card-title')}
      {renderText(text, 'card-text')}
      {children}
    </div>
  )
}

Card.propTypes = {
  children: array,
  classes: array,
  onClick: func.isRequired,
  text: string,
  title: string,
}

export default Card
