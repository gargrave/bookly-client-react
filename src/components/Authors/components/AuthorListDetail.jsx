import React from 'react'
import { func, number, shape, string } from 'prop-types'

import './AuthorListDetail.css'

const AuthorListDetail = props => {
  return (
    <div className="author-list-detail" onClick={props.onClick}>
      <p>
        {props.author.firstName} {props.author.lastName}
      </p>
    </div>
  )
}

AuthorListDetail.propTypes = {
  author: shape({
    id: number.isRequired,
    firstName: string.isRequired,
    lastName: string.isRequired,
    createdAt: string.isRequired,
    updatedAt: string.isRequired
  }),
  onClick: func.isRequired
}

export default AuthorListDetail
