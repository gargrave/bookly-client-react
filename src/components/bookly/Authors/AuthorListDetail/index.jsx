// @flow
import React from 'react';
import { func, shape, string } from 'prop-types';

import type { Author } from '../../../../constants/flowtypes';

import Card from '../../../common/Card';

type Props = {
  author: Author,
  onClick: Function,
};

function AuthorListDetail({
  author,
  onClick,
}: Props) {
  return (
    <Card
      classes={['author-card']}
      onClick={onClick}
      title={`${author.firstName} ${author.lastName}`}
    />
  );
}

AuthorListDetail.propTypes = {
  author: shape({
    firstName: string.isRequired,
    lastName: string.isRequired,
  }),
  onClick: func.isRequired,
};

export default AuthorListDetail;
