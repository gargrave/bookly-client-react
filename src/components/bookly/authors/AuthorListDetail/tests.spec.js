import React from 'react'
import { shallow } from 'enzyme'

import { authorsMockData } from '@/utils/mocks/data/authors-mock-data'

import Card from '@/components/common/Card'

import AuthorListDetail from './'

describe('AuthorListDetail', () => {
  let props
  let component

  beforeEach(() => {
    props = {
      author: Object.create(authorsMockData[0]),
      onClick: jest.fn(),
    }

    component = shallow(<AuthorListDetail {...props} />)
  })

  test('renders correctly', () => {
    const element = component.find(Card)
    expect(component).toMatchSnapshot()
    expect(element.length).toEqual(1)
  })
})
