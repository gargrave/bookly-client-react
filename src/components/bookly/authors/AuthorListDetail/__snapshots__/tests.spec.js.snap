// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorListDetail renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthorListDetail
    author={Object {}}
    onClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Array [
        "author-card",
      ],
      "onClick": [Function],
      "title": "Aldous Huxley",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Array [
          "author-card",
        ],
        "onClick": [Function],
        "title": "Aldous Huxley",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
